Sources/xcodeproj/Project/XcodeProj.swift

```diff
@@ -69,6 +75,26 @@ public final class XcodeProj: Equatable {
             lhs.pbxproj == rhs.pbxproj &&
             lhs.sharedData == rhs.sharedData
     }
+    
+    // MARK: - Private
+    
+    private static func readPBXProj(path: Path) throws -> (Data, [String: Any]) {
+        let plistXML = try Data(contentsOf: path.url)
+        var propertyListFormat =  PropertyListSerialization.PropertyListFormat.xml
+        let serialized = try PropertyListSerialization.propertyList(
+            from: plistXML,
+            options: .mutableContainersAndLeaves,
+            format: &propertyListFormat
+        )
+        guard let dictionary = serialized as? [String: Any] else {
```

Comment:
You can safely cast the `serialised` object into a [String: Any] with `as!`. I don't see Xcode replacing the root format of the `pbxproj` file.

Reply:
I had to add an extension. Method `enumerateKeysAndObjects` doesn't throw error.